designs/disruption-controls.md

```diff
@@ -0,0 +1,118 @@
+# Disruption Controls - (in v1beta1+)
+
+## Motivation
+Users have been asking for more control around the speed of Consolidation and the ability to restrict disruptions to their nodes. The demand for this is reflected by the +1s in [#1738](https://github.com/aws/karpenter/issues/1738). This is increasingly important as clusters can run at a highly varied scales, where some users have reported scale-down behaving too slow at higher scales, or too quick at smaller scales. Additionally, users that want to control when nodes can be disrupted must add PDBs to block node disruptions, add `do-not-evict` annotations on pods for sensitive workloads, or add `do-not-consolidate` annotations on nodes that shouldn't be consolidated. Natively integrating a better way to ensure nodes aren't disrupted will ease the burden that users have to take on with disruption.
```

Comment:
From experience with a controller that did rolling updates under kOps:

We have time-restricted automated node drift/expiration remediation to business hours so that there are personnel around to deal with any issues caused by the disruption.

We currently limit concurrent node replacements to 30. That value is somewhat arbitrary and has not been tuned. With a value much higher than that we encountered problematic behavior stemming from excessive load on the Kubernetes API.